Authorization/Authorization/Presentation/SocialSign/SocialSignView.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  SocialSignView.swift
+//  Authorization
+//
+//  Created by Eugene Yatsenko on 10.10.2023.
+//
+
+import SwiftUI
+import Core
+
+struct SocialSignView: View {
+
+    // MARK: - Properties
+
+    @StateObject var viewModel: SocialSignViewModel
+
+    init(
+        signType: SignType = .signIn,
+        viewModel: SocialSignViewModel
+    ) {
+        self._viewModel = .init(wrappedValue: viewModel)
+        self.signType = signType
+    }
+
+    enum SignType {
+        case signIn
+        case register
+    }
+    var signType: SignType = .signIn
+
+    private var title: String {
+        switch signType {
+        case .signIn:
+            return AuthLocalization.signInWith
```

Comment:
In the latest xcode, the return is redundant, and it's also being used at multiple places in the codebase. So you can skip the return here.